lib/vega_lite/export.ex

```diff
@@ -85,4 +102,96 @@ defmodule VegaLite.Export do
   defp escape_double_quotes(json) do
     String.replace(json, ~s{"}, ~s{\\"})
   end
+
+  @doc """
+  Renders the given graphic as a PNG image and returns
+  its binary content.
+
+  Relies on the `npm` packages mentioned above.
+  """
+  @spec to_png(VegaLite.t()) :: binary()
+  def to_png(vl) do
+    node_convert(vl, "png", "to_png/1")
+  end
+
+  @doc """
+  Renders the given graphic as an SVG image and returns
+  its binary content.
+
+  Relies on the `npm` packages mentioned above.
+  """
+  @spec to_svg(VegaLite.t()) :: binary()
+  def to_svg(vl) do
+    node_convert(vl, "svg", "to_svg/1")
+  end
+
+  @doc """
+  Renders the given graphic into a PDF and returns its
+  binary content.
+
+  Relies on the `npm` packages mentioned above.
+  """
+  @spec to_pdf(VegaLite.t()) :: binary()
+  def to_pdf(vl) do
+    node_convert(vl, "pdf", "to_pdf/1")
+  end
+
+  defp node_convert(vl, format, fn_name) do
+    json = to_json(vl)
+    json_file = System.tmp_dir!() |> Path.join("vl.json")
```

Comment:
I am afraid this will fail under concurrent usage, one vl.json file can override the other. Maybe instead of using System.cmd, we should use `Port.open` directly and stream the json to STDIN? Or at least make the path above random.

Reply:
Good point! Initially I thought we should use STDIN, but if I understand this correctly we cannot pass STDIN, then EOF (without closing the port) and retrieve all the output. At least without [external dependencies](https://github.com/jayjun/rambo).

Reply:
I see. This should be enough for a random file then: "#{:erlang.phash2({node(), self()})}-#{System.os_time()}"
"103533917-1621865213452485000"